source/configure-connection.txt

```diff
@@ -0,0 +1,175 @@
+the database settings. You must configure the dictionary's
+``default`` key to set MongoDB as the default database
+connection.
+
+To configure the ``default`` key, assign a nested dictionary as its value.
+This nested dictionary has the following keys:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 80
+
+   * - Key
+     - Description
+
+   * - **ENGINE**
+     - The backend driver to use for the connection. Set this key to ``"django_mongodb_backend"``.
+
+   * - **HOST**
+     - | Your connection URI. For localhost connections, this key is optional. 
+       | For SRV connections, you must include a scheme prefix (``mongodb+srv://``).
+       |
+       | If connecting to a replica set or sharded cluster with multiple hosts, specify 
+       | each host separated by a comma.
+
+   * - **NAME**
+     - The database you want to use.
+
+   * - **USER**
+     - The username for authenticating to the database.
+
+   * - **PASSWORD**
+     - The password for your database user.
+
+   * - **PORT**
+     - | The port number on which the database server is listening. 
+       | For MongoDB Atlas connections, this key is optional.
+
+   * - **OPTIONS**
+     - A dictionary of additional connection options for the database. This key is optional.
+
+.. _django-manual-config-example:
+
+Example 
+```````
+
+This example specifies the ``DATABASES`` variable to connect
+to a MongoDB deployment with the following configuration:
+
+- Connects to the ``my_database`` database
+- Provides authentication information for a database user
+  whose username is ``my_user`` and password is ``my_password``
+- Uses the default MongoDB port (``27017``)
+- Sets the ``retryWrites`` connection option to ``true``,
+  which configures the driver to automatically retry certain
+  write operations if they fail
+- Sets the ``w`` connection option to ``majority``,
+  which configures the driver to wait for acknowledgement from a majority
+  of replica set members before performing write operations
+
+.. code-block:: python
+
+    DATABASES = {
+        "default": {
+            "ENGINE": "django_mongodb_backend",
+            "HOST": "mongodb+srv://cluster0.example.mongodb.net",
+            "NAME": "my_database",
+            "USER": "my_user",
+            "PASSWORD": "my_password",
+            "PORT": 27017,
+            "OPTIONS": {
+                "retryWrites": "true",
+                "w": "majority",
+            },
+        },
+    }
+
+.. tip::
+
+    To see a full list of connection options that you
+    can set in the ``OPTIONS`` key, see the optional
+    parameters for `MongoClient <https://pymongo.readthedocs.io/en/4.10.1/api/pymongo/mongo_client.html#pymongo.mongo_client.MongoClient>`__
+    in the PyMongo API documentation.
+
+.. _django-connection-configure-automatic:
+
+Automatically Configure Database Settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To automatically construct the ``DATABASES`` setting that configures
```

Comment:
q: does this literally construct the DATABASES variable in that file with that same structure? or does it just have the same effect on the client/connection?

Reply:
The Django MongoDB README says it "constructs a DATABASES setting equivalent to the first example" -- @Jibola is it accurate to say that parse_uri() automatically constructs this setting?

Reply:
`parse_uri` is a convenience method so if it helps to leave it out for clarity's sake then I would do that. That said, `DATABASES` has to exist and it has to have the required keys. For example, I can run Django without that setting but if I try to access the admin I get
```
settings.DATABASES is improperly configured. Please supply the ENGINE value. 
Check settings documentation for more details.
```

Also FWIW my preference for testing is currently: https://github.com/aclark4life/project-templates/blob/main/project_template/project_name/settings/__init__.py#L7 with `MONGODB_URI` set by direnv.

Reply:
It seems accurate to say it's constructing the DATABASES setting, so I'll leave that in. 
Since I'm showing how to configure the same settings as the previous example, I'll leave the MONGODB_URI variable that shows how to do that